server/src/db/repositories/score-repo.ts

```
@@ -0,0 +1,195 @@
+import { Score } from '../models/score';
+import { User } from '../models/user';
+
+interface IUserInfo {
+  email: string;
+  name: string;
+  img: any;
+}
+interface IUserScoresData {
+  totalScores: number;
+  userRank: number | null;
+  userFounded: boolean;
+  data: [];
+}
+
+export function updateUserScore(
+  userId: string,
+  modelId: string,
+  score: number,
+) {
+  return Score.findOne({
+    modelId,
+    userId,
+  })
+    .then((userScore: any) => {
+      if (userScore && userScore.score < score) {
```

With this logic, should the function name be `updateUserHighScore`? Otherwise I'd assume that I can call it with any score and it'll get captured. As it's written, if I give it a lower score it's silently ignored, which is a little confusing.